File: Bai_52/main.py
import math

def sum_of_divisors(n):
    sum_divisors = 0
    for i in range(1, int(math.sqrt(n)) + 1):
        if n % i == 0:
            sum_divisors += i
            if n // i != i:
                sum_divisors += n // i
    return sum_divisors

if __name__ == "__main__":
    with open("TONGUOC.INP", "r") as ifile, open("TONGUOC.OUT", "w") as ofile:
        K, Q = map(int, ifile.readline().split())
        for _ in range(Q):
            L, R = map(int, ifile.readline().split())
            cnt = 0
            for j in range(L, R + 1):
                if sum_of_divisors(j) >= K:
                    cnt += 1
            ofile.write(str(cnt) + "\n")

File: Bai_52/main.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>
using namespace std;

int sumOfDivisors(int n)
{
    int sum = 0;
    for(int i = 1; i <= sqrt(n); i++)
    {
        if(n % i == 0)
        {
            sum += i;
            if(n / i != i) // Tránh trường hợp n là bình phương của một số
                sum += n / i;
        }
    }
    return sum;
}

int main() {
    
    ifstream ifile("TONGUOC.INP");
    ofstream ofile("TONGUOC.OUT");

    int K, Q;
    ifile >> K >> Q;
    while (Q--)
    {
        int L, R;
        ifile >> L >> R;
        int cnt = 0;
        for (int j = L; j <= R; j++) 
            if (sumOfDivisors(j) >= K) 
                cnt ++;
        ofile << cnt << endl;
    }
    ifile.close();
    ofile.close();

    return 0;
}